File: packages/python-sdk/src/services/contract_service.py

from web3 import Web3
from eth_typing import Address
from typing import Any

class ContractService:
    def __init__(self, provider: Web3, account: Address):
        self.provider = provider
        self.account = account

    async def validate_network(self, expected_chain_id: int) -> None:
        network = await self.provider.eth.chain_id
        if network != expected_chain_id:
            raise Exception(f"Chain ID mismatch. Expected {expected_chain_id}, got {network}")

    def create_contract(self, address: str, abi: Any) -> Any:
        return self.provider.eth.contract(address=address, abi=abi)

File: packages/sdk/src/services/ContractService.ts
import { ethers } from "ethers";

export class ContractService {
  private provider: ethers.Provider;
  private signer: ethers.Signer;
  
  constructor(provider: ethers.Provider, signer: ethers.Signer) {
    this.provider = provider;
    this.signer = signer;
  }

  async validateNetwork(expectedChainId: number): Promise<void> {
    const network = await this.provider.getNetwork();
    if (network.chainId !== BigInt(expectedChainId)) {
      throw new Error(`Chain ID mismatch. Expected ${expectedChainId}, got ${network.chainId}`);
    }
  }

  createContract(address: string, abi: any): ethers.Contract {
    return new ethers.Contract(address, abi, this.signer);
  }
} 